@fast-check/vitest cannot find module

## 🐛 Bug Report

`vitest-fast-check.js` cannot find an internal module it needs to function. `fast-check` itself works perfectly fine in `vitest`, so I wonder if it's a problem with my own import system. Sorry if this issue isn't relevant currently because `@fast-check/vitest` is so new.

Error:

```sh
Error: Cannot find module '/path/to/project/node_modules/.pnpm/@fast-check+vitest@0.0.2_vitest@0.28.5/node_modules/@fast-check/vitest/lib/esm/internals/TestBuilder' imported from /path/to/project/node_modules/.pnpm/@fast-check+vitest@0.0.2_vitest@0.28.5/node_modules/@fast-check/vitest/lib/esm/vitest-fast-check.js
```

## To Reproduce

Steps to reproduce:

- Run `pnpm i -D @fast-check/vitest` to install
- Create test:

```ts
// ./test/sanity.test.ts
import { fc, test } from "@fast-check/vitest";

test.prop([fc.string()])("Passing sanity test", (s) => {
  const testing = s === s;
  console.log(testing)
};
```

- Run `pnpm test` (which runs `pnpm vitest run`)

## Expected behavior

Library should find file imported from `@fast-check/.../internals/TestBuilder`

## Your environment

| Packages / Softwares | Version(s) |
| -------------------- | ---------- |
| fast-check                 | ^3.6.3       |
| @fast-check/vitest   | ^0.0.2       |
| node                           | 18.7.0        |
| TypeScript\*              | 4.9.5         |

\*Only for TypeScript's users

--- a/.yarn/versions/66290b37.yml
+++ b/.yarn/versions/66290b37.yml
@@ -1,0 +1,2 @@
+releases:
+  "@fast-check/vitest": patch

--- a/packages/vitest/package.json
+++ b/packages/vitest/package.json
@@ -24,7 +24,7 @@
     "build:publish-cjs": "tsc -p tsconfig.publish.json",
     "build:publish-esm": "tsc -p tsconfig.publish.json --module es2015 --moduleResolution node --outDir lib/esm && cp package.esm-template.json lib/esm/package.json",
     "test": "jest",
-    "test-bundle": "vitest --config test-bundle/vitest.config.mjs",
+    "test-bundle": "node test-bundle/basic.mjs && vitest --config test-bundle/vitest.config.mjs",
     "typecheck": "tsc --noEmit"
   },
   "repository": {

--- a/packages/vitest/src/vitest-fast-check.ts
+++ b/packages/vitest/src/vitest-fast-check.ts
@@ -1,6 +1,6 @@
 import { it as itVitest, test as testVitest } from 'vitest';
 import * as fc from 'fast-check';
-import { buildTest } from './internals/TestBuilder';
+import { buildTest } from './internals/TestBuilder.js';
 
 import type { FastCheckItBuilder } from './internals/TestBuilder';
 import type { It } from './internals/types';

--- a/packages/vitest/test-bundle/basic.mjs
+++ b/packages/vitest/test-bundle/basic.mjs
@@ -1,0 +1,25 @@
+// Checking that our package can be properly imported by ES Module imports.
+// It seems that in some cases, Vitest allow imports of invalid packages, with that dedicated
+// test case, we make sure that the package is correct from a Node point of view.
+
+import('@fast-check/vitest').then(
+  () => {
+    // In theory, Vitest is supposed to throw when imported outside of its own execution context
+    // but we prefer not failing if it starts not to be the case anymore as it sounds rather like
+    // testing an implementation detail of Vitest.
+  },
+  (err) => {
+    if (
+      err.message.includes('Vitest was initialized with native Node instead of Vite Node') &&
+      err.stack.includes('at new VitestUtils')
+    ) {
+      // When Vitest throws because it was not executed in the right context we receive: raw Error thrown by VitestUtils.
+      // We ignore errors linked to Vitest not being executed into the right context as this file is supposed not to be executed
+      // with the right context so it's fully expected to throw at this point.
+      return;
+    }
+    // When our modules are not properly defined, we receive: err.code === "ERR_MODULE_NOT_FOUND".
+    // Or at least, it's one of the possible errors we could receive.
+    throw err;
+  }
+);

